streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamAggregate.java

```diff
@@ -142,7 +142,7 @@ public void process(final Record<KIn, VIn> record) {
         private TimestampedKeyValueStore<KIn, VAgg> store;
 
         @Override
-        public void init(final org.apache.kafka.streams.processor.ProcessorContext context) {
+        public void init(final ProcessorContext<?, ?> context) {
```

Comment:
Interesting. As far as I can see the value getters do not forward, so retricting the types could be an option. 
At the moment though, these value getters are initialized as part of the processor itself, and changing this types would break things.

Do you think we should consider redesign this internal api inthe future as well?